cleanlab/dataset.py

```diff
@@ -0,0 +1,416 @@
+    num_overlapping = (df["Joint Probability"] * num_examples).round().astype(int)
+    df.insert(loc=2, column="Num Overlapping Examples", value=num_overlapping)
+    if class_names is not None:
+        df.insert(
+            loc=0, column="Class Name A", value=df["Class Index A"].apply(lambda x: class_names[x])
+        )
+        df.insert(
+            loc=1, column="Class Name B", value=df["Class Index B"].apply(lambda x: class_names[x])
+        )
+    return df.sort_values(by="Joint Probability", ascending=False)
+
+
+def overall_label_health_score(
+    labels=None,
+    pred_probs=None,
+    *,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+    verbose=True,
+):
+    """Returns a single score/metric between 0 and 1 for the quality of all labels in a dataset.
+    Intuitively, the score is the average label consistency across all classes in the
+    dataset. So a score of 1 suggests your data is perfectly labeled and a score of 0.5 suggests
+    that, on average across all classes, about half of the label may have issues. Thus, a higher
+    score implies higher quality labels, with 1 implying labels that have no issues.
+
+    Parameters
+    ----------
+    For parameter info, see the docstring of `dataset.find_overlapping_classes`
+
+    Returns
+    -------
+    health_score : float
+        A score between 0 and 1 where 1 implies the dataset has all estimated perfect labels.
+        A score of 0.5 implies that, on average, half of the dataset's label have estimated issues.
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+    joint_trace = joint.trace()
+    if verbose:
+        num_issues = (num_examples * (1 - joint_trace)).round().astype(int)
+        print(
+            f" * Overall, about {1 - joint_trace:.0%} ({num_issues:,} of the {num_examples:,}) "
+            f"labels in your dataset have issues.\n"
+            f" ** The overall label health score for this dataset is: {joint_trace:.2f}."
+        )
+    return joint_trace
+
+
+def health_summary(
+    labels=None,
+    pred_probs=None,
+    *,
+    asymmetric=False,
+    class_names=None,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Prints a healthy summary of your datasets including results for powerful statistics like:
+    * the classes with the most and least label issues
+    * classes that overlap and could potentially be merged
+    * overall data label quality health score statistics for your dataset
+
+    Parameters
+    ----------
+    For parameter info, see the docstring of `dataset.find_overlapping_classes`
```

Comment:
I'd imagine `health_summary()` is the method we envision most users invoking from this module (and just getting results from find_overlapping_classes from the internal call inside health_summary).

So consider moving the docstrings for parameters to this method, and then having the Docstring for find_overlapping_classes say: See docstring of `health_score` instead.